include/eve/algo/ptr_iterator.hpp

```diff
@@ -89,6 +92,14 @@ namespace eve::algo
     auto unaligned() const { return unaligned_ptr_iterator<T, Cardinal>{ptr.get()}; }
     auto previous_partially_aligned() const { return *this; }
 
+    template <typename _Cardinal>
+    auto cardinal_cast(_Cardinal) const
```

Comment:
is it something we don't want as a free function for any iterator type ?

Reply:
It's a method that we need for every iterator implemented separately.

Reply:
OK but why as a member function ?

Reply:
As oppose to? You mean as a friend?

it's the same as `unaligned` or `previously_partially_aligned` or what have you.

Reply:
ok